debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/converters/sink/SinkDocument.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb.converters.sink;
+
+import java.util.Optional;
+
+import org.bson.BsonDocument;
+
+public final class SinkDocument implements Cloneable {
+    private final BsonDocument keyDoc;
+    private final BsonDocument valueDoc;
+
+    public SinkDocument(final BsonDocument keyDoc, final BsonDocument valueDoc) {
+        this.keyDoc = keyDoc;
+        this.valueDoc = valueDoc;
+    }
+
+    public Optional<BsonDocument> getKeyDoc() {
+        return Optional.ofNullable(keyDoc);
+    }
+
+    public Optional<BsonDocument> getValueDoc() {
+        return Optional.ofNullable(valueDoc);
+    }
```

Comment:
I am not opposed to the use of `Optional`, but I don't care for the repeated creation of the `Optional` temporary objects for the sole purpose of nullability checks.  

If we can move the creation of the `Optional` to the constructor, then that'd be great.  If that presents an issue, then I think I'd prefer we use `!= null` or `Objects.isNull` instead.

Reply:
I'm actually in favour of using Optional. However I agree with @Naros -- just make the field Optional right away. I know IDE is going to complain, I know the language designers argued it should not be done... And it's being done anyway since they never actually managed to explain properly why and IMHO it makes more sense.

Reply:
@Naros  What's the difference in numbers of `Optional` instances? Isn't the current implementation better because the lifespan is the lifespan of the variable that calls the `get*Doc()` methods? This can be garbage collected early. When the Optional is bound to the instance of `SinkDocument` then the lifespan might be pretty much longer, no?
This is why I think the authors suggest `Optional` to be used for return-methods only and not for member-vars and that's why IDEs should complain.

Reply:
Being concerned about 2 extra references in an application such as Debezium seems a shining example of premature optimisation. I wouldn't even use the fact that the on call object construction is probably worse than the additional memory consumption. It's imho just a matter of readability TBH.

Reply:
Then I hope we can leave it like this for now? @jpechane ?
It would make my feelings better and my IDE's as well. ;)

Reply:
We do use `Optional` for attributes and from my PoV it is good defensive programming approach.

Reply:
The rest of the world AND the authors for `Optional` do not agree though. Please file a Jira or change it yourself, I can't do it because to me it just does not feel right at all for the above reasons.

Reply:
Code changed, issue resolved